### PR TITLE
Fix the output line field cleanup in case of ending colon

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/filesystem/gpfs.py [hunk 1]
```diff
@@ -73,7 +74,7 @@ def clean(line):
         else:
             return line
 
-    return [[percentdecode(y) for y in clean(x).strip().split(':')] for x in out.strip().split('\n')]
+    return [[percentdecode(y) for y in clean(x).strip().split(':')] for x in out]
```
Comment (stdweird) [inline lib/vsc/filesystem/gpfs.py:77]: what happened with the `.strip` part?

Comment (itkovian) [reply lib/vsc/filesystem/gpfs.py:76]: splitlines does not need it

Comment (itkovian) [reply lib/vsc/filesystem/gpfs.py:76]:
~~~~
>>> "ghd\ndd\n".splitlines()
['ghd', 'dd']
~~~~

Comment (stdweird) [reply lib/vsc/filesystem/gpfs.py:76]: great. didn't know that

Comment (itkovian) [reply lib/vsc/filesystem/gpfs.py]: Neither did I :)

File: test/gpfs.py [hunk 2]
```diff
@@ -37,68 +37,68 @@ def test_split_output_lines(self):
...
+            "and:here:is:line:1",
+            "and:here:is:line:2",
+            "and:here:is:line:3",
+            "and:here:is:line:4",
+        ]
 
         split_lines = gpfs.split_output_lines(test_lines)
 
-        self.assertTrue(len(set(map(len, split_lines))) == 1)  # all lines have the same number of fields
+        self.assertEqual(list(map(len, split_lines)), [6, 5, 5, 5, 5])   # all lines have the same number of fields
```
Comment (kwaegema) [inline test/gpfs.py:67]: comment is not true here? :)

**Comment by kwaegema** [comment]
LGTM

**Comment by wpoely86** [comment]
I would add a test for the output of mmhealth. All other GPFS commands seem to always return the same number of 'columns' for the `-Y` output. Only mmhealth does something special with 2 headers of different length and rows belong to either one or the other.

**Comment by wpoely86** [comment]
I've used this PR to apply all VO stuff: no issues found!

File: test/gpfs.py (hunk 2 above)
Comment (itkovian) [reply test/gpfs.py:67]: The code is the documentation :p

**Comment by itkovian** [comment]
@wpoely86 imho, we need an actual parser for this kind of stuff that creates a specific data structure for each "type" of line.

**Comment by wpoely86** [comment]
Probably but for now, this is the only 'weird' case that I've found. Everything else nicely outputs an equal number of columns for every row.

**Comment by kwaegema** [comment]
test_get_mmhealth parses mmhealth output, albeit only parts of it. there are actually very few tests for the gpfs module overall, and by making the lustre module i already found some things in the gpfs code that could never have worked. but that's for another PR, merging in :smile:

**Pull request merged**